frontend/src/components/billing/billing.tsx

```diff
@@ -0,0 +1,93 @@
+import { AppRoute, DataStatus, NotificationMessage } from 'common/enums/enums';
+import { FC, UserGetResponseWithMoneyBalanceDto } from 'common/types/types';
+import { Button, Spinner } from 'components/common/common';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useNavigate,
+} from 'hooks/hooks';
+import { Token } from 'react-stripe-checkout';
+import { notification } from 'services/services';
+import { billingActions } from 'store/actions';
+
+import { DEFAULT_REPLENISH_AMOUNTS, REPLENISH_PUBLIC_KEY } from './common';
+import { ReplenishCardsList } from './components/components';
+import styles from './styles.module.scss';
+
+const MINIMAL_AMOUNT_OF_MONEY_TO_WITHDRAW = 1;
+
+const Billing: FC = () => {
+  const { authDataStatus, user, billingDataStatus, userWithMoneyBalance } =
+    useAppSelector(({ auth, billing }) => ({
+      authDataStatus: auth.dataStatus,
+      user: auth.user,
+      billingDataStatus: billing.dataStatus,
+      userWithMoneyBalance: billing.userWithMoneyBalance,
+    }));
+
+  const navigate = useNavigate();
+
+  if (!user) {
+    navigate(AppRoute.ROOT);
+  }
+
```

Comment:
Maybe let's navigate to SIGN_IN?
Also, move this check down, after all used hooks, to avoid errors, like `rendered fewer hooks than expected`

Reply:
Previos time on Chat functionality QAs asked to make redirection to Dashboard. About errors, strange, but this time I did not get it 🤔

Reply:
Oh, I know, why I did not get an error. Navigate is an instance of hook